stock_owner_restriction/models/stock_quant.py

```diff
@@ -25,20 +26,33 @@ def _gather(
             owner_id=owner_id,
             strict=strict,
         )
-        restricted_owner_id = self.env.context.get("force_restricted_owner_id", None)
-        if owner_id is None or restricted_owner_id is None:
+        restricted_owner = self.env.context.get("force_restricted_owner_id", None)
```

Comment:
@AungKoKoLin1997 I am wondering why this context key has been added. As there is one in stock module: "owner_id".

IMHO, if we want to force the owner, we should use that one instead. This has the advantage of removing a lot of not necessary code.

Could you explain ? Thanks

Reply:
> @AungKoKoLin1997 I am wondering why this context key has been added. As there is one in stock module: "owner_id".

@rousseldenis I was also thinking about this with no definite answer. My tentative interpretation is to avoid triggering unexpected behaviors by sharing the key with the standard logic, but I'm not sure it's a valid concern.

@sergio-teruel Would be able to explain how you decided to use the `force_restricted_owner_id` context key?

Reply:
@rousseldenis 
In my opinion, adding force_restricted_owner_id alongside owner_id is a strategic choice for improving how we manage stock. owner_id simply identifies the owner, but force_restricted_owner_id gives us a nuanced way to control actions based on ownership, enhancing security. Plus, keeping these keys separate avoids confusion with other parts of the system using owner_id.